### ISSUE TITLE
Early program termination caused by disappearing g_labels

### ISSUE THREAD

**Issue by jinweixia03** [cliffbb/OEM-Fewshot-Challenge#3]
Hello, I downloaded the data set, pre-trained weights and your code, and modified the two parameters data_root and pretrained in strict accordance with the operating requirements. But some problems still occurred. I could not obtain other result information except preds. After preliminary debug investigation, I found that it exited the subsequent program because there was no q_labels, but the val_loader contained the information of images and labels. Hope to get your help.
oem.yaml
```
DATA:
  data_name: OpenEarthMap few-shot challenge
  data_root:  /home/alian/dataset/valset/valset (copy) # Download the 'valset' and set your data_root to 'valtset' folder
  images_dir: images/  
  labels_dir: labels/
  data_list_path: data/val.json
  novel_classes:   # Change the novel classnames for Validation/Testing. The current classnames are Validation classes
    8: 'road type 2'        
    9: 'river'               
    10: 'boat & ship'       
    11: 'agric land type 2'       
  workers: 8
  image_size: 1024
  mean: [0.4449, 0.4570, 0.3919] 
  std: [0.1662, 0.1492, 0.1505] 
  pin_memory: True

MODEL:
  model_name: PSPNet
  encoder_name: efficientnet-b4
  pretrained:  model.best.pth # model.best.pth

EVALUATION:
  shot: 5   # Only 5-shot is allowed
  save_pred_maps: True
  save_ious: True  
  batch_size_val: 1  # Only batch size 1 is supported (Evaluate one query image at a time)
  shuffle_test_data: False
  manual_seed: 2023
  gpus: [0]

CLASSIFIER:   # All the settings here are from the baseline paper
  weights: [100, 1, 1, 100]  
  adapt_iter: 100 
  cls_lr: 0.00125   
  pi_estimation_strategy: self
  pi_update_at: [10]
  fine_tune_base_classifier: True
```

src/test.py line156-157
```  
if q_label is None:
        return
```

**Comment by cliffbb** [comment]
@scorpio1121 

> I found that it exited the subsequent program because there was no q_labels, but the val_loader contained the information of images and labels.

**Comment by cliffbb** [comment]
@scorpio1121 

> I found that it exited the subsequent program because there was no q_labels, but the val_loader contained the information of images and labels.

It is only the `support_set`  that consists of  `images` and their `labels`,  and that is what the loader contains. You are to predict the `labels` for the `images` in the `query_set` and submit your predictions for evaluation

**Comment by cliffbb** [comment]
@scorpio1121 

> I found that it exited the subsequent program because there was no q_labels, but the val_loader contained the information of images and labels.

It is only the `support_set`  that consists of  `images` and their `labels`,  and that is what the loader contains. You are to predict the `labels` for the `images` in the `query_set` and submit your predictions for evaluation

Hope this will help if you haven't resolved it yet. Thank you

**Comment by jinweixia03** [comment]
@cliffbb 
Yes, I misunderstood the description of the dataset. Thank you so much.
